clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -431,6 +444,96 @@ private static int parseAcks(String acksString) {
     }
 
     /**
+     * Needs to be called before any other methods when the transactional.id is set in the configuration.
+     *
+     * This method does the following:
+     *   1. Ensures any transactions initiated by previous instances of the producer
+     *      are completed. If the previous instance had failed with a transaction in
+     *      progress, it will be aborted. If the last transaction had begun completion,
+     *      but not yet finished, this method awaits its completion.
+     *   2. Gets the internal producer id and epoch, used in all future transactional
+     *      messages issued by the producer.
+     *
+     * @throws IllegalStateException if the TransactionalId for the producer is not set
+     *         in the configuration.
+     */
+    public void initTransactions() {
+        if (transactionState == null || !transactionState.isTransactional()) {
+            throw new IllegalStateException("Cannot call initTransactions without setting a transactional id.");
+        }
+        FutureTransactionalResult result = transactionState.initializeTransactions();
```

Comment:
I agree. I was just emulating the pattern of `ProduceRequestResult` and `FutureRequestMetadata`. We can simplify this one though.